### DISCUSSION TITLE
DMD or LDC Docker Setup for Mac M1 - Any Success Stories or Recommendations?

### DISCUSSION THREAD

**Discussion by Ananiaslitz** [dlang/dmd#15655] [Q&A]
Hello D community,

I hope everyone is doing well. I've been working on setting up a Docker environment for a D language project on a Mac with the M1 chip. I've encountered some challenges, particularly when trying to get the DMD and LDC compilers to work within a Docker container on this platform.

I was wondering if anyone has successfully set up a Docker image with either the DMD or LDC compiler that is compatible with the Mac M1 architecture? If so, could you please share any insights or resources on how to achieve this?

Thank you in advance for your assistance and guidance.

**Comment by tom-tan** [comment]
You can use [install.sh](https://dlang.org/install.html) to install LDC.
It works on arm64 and amd64.
Unfortunately, it does not work for dmd for M1 architecture because the dmd binary for arm64 is not officially available.
